### ISSUE TITLE
Jest Projects: Could not find 'jest-dynalite-config.js' file

### ISSUE THREAD

**Issue by Purii** [freshollie/jest-dynalite#6]
Hi!

Thanks for that awesome package!
Unfortunately my setup doesn't seem to work with `v2` anymore.
I get the following error:

```
Test suite failed to run

    Could not find 'jest-dynalite-config.js' file

      at loadConfig (../../../node_modules/jest-dynalite/dist/config.js:12:11)
      at Object.<anonymous> (../../../node_modules/jest-dynalite/dist/config.js:15:14)
      at Object.<anonymous> (../../../node_modules/jest-dynalite/dist/setup.js:3:18)
```

--> Where does `jest-dynalite` look for the config file?

```js
|-jest.config.js
|-functions
|   |-jest.config.js
|   |-jest-dynalite-config.js
|   |-jest.setupAfterEnv.js
|   |-jest.setupBeforeEnv.js

// jest.config.js
module.exports = {
  projects: [
    "<rootDir>/functions/jest.config.js",
  ],
};

// functions/jest.config.js
module.exports = {
  name: "functions",
  displayName: "functions",
  testMatch: ["**/*.test.ts", "**/test.ts"],
  preset: "jest-dynalite",
  setupFiles: ["./jest.setupBeforeEnv.js"],
  setupFilesAfterEnv: ["./jest.setupAfterEnv.js"],
  testTimeout: 10000,
  transform: {
    "^.+\\.tsx?$": "ts-jest",
  },
};

// functions/jest.setupBeforeEnv.js
const { setup } = require("jest-dynalite");

setup(__dirname);
```

**Comment by David-Tennant** [comment]
@Purii Howdy. There are two issues that I can see

### Two jest config files

```
|-jest.config.js // The closest file to the root is probably going to run
|-functions
|   |-jest.config.js // Your example is this
|   |-jest-dynalite-config.js
|   |-jest.setupAfterEnv.js
|   |-jest.setupBeforeEnv.js
```

Do you have two config files? Which one is Jest using?
Make sure you are choosing the right one
https://jestjs.io/docs/en/configuration

### Jests path resolution gotcha

A little while back I had a similar issue with some other tools. It may have something to do with where your config file is being run from: `Default: The root of the directory containing your Jest config file or the package.json or the pwd if no package.json is found`

Specifying the root path for your file resolutions: https://jestjs.io/docs/en/configuration#rootdir-string

This
```
setupFilesAfterEnv: ["./jest.setupAfterEnv.js"],
```
vs
```
setupFilesAfterEnv: ["<rootDir>/functions/jest.setupAfterEnv.js"],
```

**Comment by Purii** [comment]
Hi @David-Tennant!

Thanks for your answer. Tried setting `<rootDir>` but didn't help.
Strange is, if I add a second `jest-dynalite-config.js` to root, I get an error, that theres already an instance running.

The setup works quite fine. These are specific jest configs for each jest project.

Is it possible, that `jest-dynalite` runs without config? Because my setup started breaking with this commit: 14d80970a86408a6d43b30ba15eff86e212346ac

Will have a look on that the next days.

**Comment by freshollie** [comment]
Ah! Are you running jest against both the "functions" and root folder? Is it possible that before your root jest.config.js was running jest-dynalite but before 14d8097 a config wasn't required. 

It's weird that you get this clashing issue. Is it always happening? What's the reason for the 2 jest configs?

**Comment by freshollie** [comment]
Thanks for flagging @Purii. I think what's going on is you are using the preset while using `setupFiles` which causes 2 instances to be created as the preset uses a jest environment which creates its own instance of `jest-dynalite`.

I'm gonna look into warning about this on startup or stopping it from happening. Possible solution is to check if dynalite is already started and not starting another instance if so?

**Comment by freshollie** [comment]
PR in for this https://github.com/freshollie/jest-dynalite/pull/7 I will release as `v2.0.2` @Purii

**Comment by Purii** [comment]
`v2.0.2` seems to fix the issue partly. Thanks for that quick release!
Works now with two copies of `jest-dynalite-config.js` (`/jest-dynalite-config.js` + `functions/jest-dynalite-config.js`.

-> Could it be that [this check (14d80)](https://github.com/freshollie/jest-dynalite/commit/14d80970a86408a6d43b30ba15eff86e212346ac) doesn't seem to work with jest projects? Because it seems that I need one copy to satisfy the check and the other copy to actually provide the config.
Need to debug which path `fs.existsSync` uses vs. `require`.

**Comment by freshollie** [comment]
Ah! I know what's going on.

`preset: "jest-dynalite"` makes jest run `setup` with `process.cwd()`, which seems to be running before your `setupFiles`. This means that #7 doesn't fix the issue I was trying to fix as the environment is created before `setupFiles` are run.

If you are using a monorepo structure like this, `preset: "jest-dynalite"` shouldn't be used, and instead apply `import "jest-dynalite/withDb";` to your `jest.setupAfterEnv.js`

**Comment by freshollie** [comment]
@Purii I am going to provide a better warning when using `preset: "jest-dynalite"` and no config file can been found:

```
jest-dynalite could not find a config file in the jest <rootDir> (${rootDir}).

If you didn't intend to be using this directory for the config, please specify a custom
directory: https://github.com/freshollie/jest-dynalite/#monorepoadvanced-setup

If you are already using a custom config directory, you should apply 'import "jest-dynalite/withDb"'
to your "setupFilesAfterEnv" instead of using the jest-dynalite preset or environment.

Please see https://github.com/freshollie/jest-dynalite/issues/6#issuecomment-654167172 for
more info.
```

**Comment by Purii** [comment]
@freshollie sorry for the silence. It works for me with two config files.
I removed `preset: "jest-dynalite"`. But it still throws if there's no config file in root: `Could not find 'jest-dynalite-config.js' file`.

`/jest-dynalite-config.js`: Can be empty. File needs to exist to prevent error
`/functions/jest-dynalite-config.js`: "Real" content

Run Jest from `/`: `jest --projects functions`

```
`setupBeforeEnv.js`
console.log(__dirname) // -> returns /functions -> Should be fine?

const { setup } = require("jest-dynalite");

setup(__dirname);
```

Is it possible that the check runs before calling `setup`? (https://github.com/freshollie/jest-dynalite/blob/master/src/config.ts#L20?)
```
    Could not find 'jest-dynalite-config.js' file

      1 | console.log(__dirname);
    > 2 | const { setup } = require("jest-dynalite");
```

**Comment by freshollie** [comment]
Ah yeah! Don't know how I missed that! Bug was introduced with #5. Gonna try and sort it out today! Apologies for the inconvenience.

**Comment by Purii** [comment]
No hurry!

**Comment by freshollie** [comment]
@Purii this should be fixed in `v3.0.0`. The major version is because the preset should now pick up the `jest-dynalite-config.js` from the same directory as your jest config. There should now be no need to use `setup(...)` in a monorepo style application.

**Comment by Purii** [comment]
Works perfectly. Thanks @freshollie!
